ios/RNCConnectionStateWatcher.m

```diff
@@ -37,10 +37,9 @@ - (void)dealloc
 {
     self.delegate = nil;
 
-    if (self.reachabilityRef != NULL) {
+    if (self.reachabilityRef != nil) {
         SCNetworkReachabilityUnscheduleFromRunLoop(self.reachabilityRef, CFRunLoopGetMain(), kCFRunLoopCommonModes);
         CFRelease(self.reachabilityRef);
-        self.reachabilityRef = nil;
```

Comment:
What was the reason behind removing this line? Do we not want to clean up the reference we hold to the reachability instance here?

Reply:
Since we're in `dealloc`, there's no need to explicitly `nil` anything unless you expect it to be used while still in the method. I don't think it's needed here.

Reply:
I don't think this is relevant with ARC, but I've put it back. I don't see any harm in keeping it.